ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -263,6 +274,15 @@ def start_learning(self, env: BaseUnityEnvironment, trainer_config):
             self._write_training_metrics()
             self._export_graph()
 
+    def end_trainer_episodes(self, env, lessons_incremented):
+        curr_info = self._reset_env(env)
+        for brain_name, trainer in self.trainers.items():
+            trainer.end_episode()
+        for brain_name, changed in lessons_incremented.items():
+            if changed:
+                self.trainers[brain_name].reward_buffer.clear()
```

Comment:
If we're doing a generalization reset, do we need to clear the reward buffers? I don't think that will currently happen.

Reply:
Confirming, no need to change reward buffers in the case of generalization reset as generalization doesn't make use of reward buffers to decide

Reply:
As discussed offline, please add a comment to this effect; it's not obvious from the code.